### PR TITLE
Conditions parser

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

**Review by aqk — APPROVED** [review]
Nice catch. I was talking with Art over the weekend about adding more type checking in the higher level ChiaLisp - we can do some data shape and value checks as types, maybe including this.

(We can introduce new types like "Condition")

**Comment by arvidn** [comment]
this PR depends on https://github.com/Chia-Network/clvm_rs/pull/117

File: tests/core/full_node/test_mempool.py [hunk 1]
```diff
@@ -2061,6 +2061,30 @@ def generator_condition_tester(
 
 
 class TestGeneratorConditions:
+    @pytest.mark.parametrize("rust_checker", [True, False])
+    def test_invalid_condition_args_terminator(self, rust_checker: bool):
+
+        # note how the condition argument list isn't correctly terminated with a
+        # NIL atom. This is allowed, and all arguments beyond the ones we look
+        # at are ignored, including the termination of the list
```
Comment (aqk) [inline tests/core/full_node/test_mempool.py:2069]: Could be a little more explicit, and say "The conditions checking code only reads up to the "50" below

File: tests/core/full_node/test_mempool.py [hunk 2]
```diff
@@ -2061,6 +2061,30 @@ def generator_condition_tester(
...
+        assert len(npc_result.npc_list[0].conditions) == 1
+        assert npc_result.npc_list[0].conditions[0][0] == opcode
+        assert len(npc_result.npc_list[0].conditions[0][1]) == 1
+        c = npc_result.npc_list[0].conditions[0][1][0]
+        assert c == ConditionWithArgs(opcode=ConditionOpcode.ASSERT_SECONDS_RELATIVE, vars=[bytes([50])])
+
+    @pytest.mark.parametrize("rust_checker", [True, False])
+    def test_invalid_condition_list_terminator(self, rust_checker: bool):
+
+        # note how the list of conditions isn't correctly terminated with a
```
Comment (aqk) [inline tests/core/full_node/test_mempool.py:2083]: Could add that the condition list `(80 50)` is properly terminated.

**Review by aqk — APPROVED** [review]

**Pull request merged**